homeassistant/components/jellyfin/__init__.py

```diff
@@ -23,17 +24,14 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     try:
         client = await validate_input(hass, entry.data)
-    except CannotConnect:
-        _LOGGER.error("Cannot connect to Jellyfin server")
-        return False
-    except InvalidAuth:
-        _LOGGER.error("Failed to login to Jellyfin server")
-        return False
-    except Exception:  # pylint: disable=broad-except
-        _LOGGER.exception(
+    except CannotConnect as ex:
+        raise ConfigEntryNotReady("Cannot connect to Jellyfin server") from ex
+    except InvalidAuth as ex:
+        raise ConfigEntryAuthFailed("Failed to login to Jellyfin server") from ex
+    except Exception as ex:  # pylint: disable=broad-except
+        raise ConfigEntryNotReady(
```

Comment:
I am not sure what would be best for a broad-except. `ConfigEntryNotReady` will retry with an exponential pattern, which would make sense if HA can recover from the exception.

It would be good to at least log the exception, since this is something that shouldn't happen.
```python3
_LOGGER.exception(ex)
```

Reply:
Thanks for your comments! Although the use of the broad exception would be unexpected, because the ConfigEntry was created we know that at that point in time the connection to the server was succesful. So I think it is fair to expect the setup to succeed at some point in time.

I will include logging the exception.

Reply:
This is done now. I had to rebase once more because #49270 was merged today, so there were additional CI errors.

Reply:
😕
`async_setup_entry` should return False unless this is normal behavior, but then the exception should be more specific. Now the integration would spam the logs with exceptions.
The integration should log once and return False.
The log should include the exception message and entry setup failed so we can identify where the exception occurred.

Reply:
Understood. I've made the change, so now the integration will only retry if there is a connectivity issue. For all other problems, it logs the exception and fails the setup.